UnityProject/Assets/Scripts/Objects/Engineering/AntiBreachingShield.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Objects.Engineering
+{
+	public class AntiBreachingShield : MonoBehaviour, ICheckedInteractable<HandApply>, INodeControl
+	{
+		private SpriteHandler spriteHandler;
+
+		public ElectricalNodeControl ElectricalNodeControl;
+		public ResistanceSourceModule ResistanceSourceModule;
+
+		public bool connectedToOther = false;
+		public float Voltage;
+		public bool IsOn { get; private set; } = false;
+
+		#region Lifecycle
+
+		private void Awake()
+		{
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+		}
+
+		#endregion Lifecycle
+
+		#region Interaction
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+			if (interaction.TargetObject != gameObject) return false;
+			if (interaction.HandObject != null) return false;
+
+			return true;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			TogglePower();
+		}
+
+		#endregion Interaction
+
+		private void TogglePower()
+		{
+			IsOn = !IsOn;
+			UpdateSprites();
+		}
+
+		public void PowerNetworkUpdate()
+		{
+			//Voltage = ElectricalNodeControl.Node.Data.ActualVoltage;
```

Comment:
this still WIP?

Reply:
thats the old script that was on the field gen, suppose I could get rid of it